DisCatSharp/Net/WebSocket/WebSocketClient.cs

```diff
@@ -296,7 +297,7 @@ internal async Task ReceiverLoopAsync()
                     if (!this._isConnected && result.MessageType != WebSocketMessageType.Close)
                     {
                         this._isConnected = true;
-                        await this._connected.InvokeAsync(this, new SocketEventArgs()).ConfigureAwait(false);
+                        await this._connected.InvokeAsync(this, new SocketEventArgs(new ServiceCollection().BuildServiceProvider())).ConfigureAwait(false);
```

Comment:
This would mean the developer has no way of retrieving the services they registered, as this passes around a default collection that doesn't have anything.

What you did for the BaseDiscordClient should be applied here. Where we can pass in an externally created ServiceProvider

Reply:
I know. At the moment I am thinking about how it would be more correct to transfer the collection here, because there is no access to BaseDiscordClient at all.